LuaRules/Gadgets/game_share_mode.lua

```diff
@@ -330,35 +420,51 @@ function gadget:RecvLuaMsg(message, playerID) -- Entry points for widgets to int
 				targetID = tonumber(targetID)
 			end
 		end
-
-		if strFind(command,"remerge") then -- remerging seems impossible gadget side.
-			local _,active,spec,_ = spGetPlayerInfo(playerID, false)
-			if controlledPlayers[playerID] and not spec then
-				spAssignPlayerToTeam(playerID, controlledPlayers[playerID])
-				spEcho("game_message: Player " .. name .. " has been remerged!")
-			end
-		elseif strFind(command,"unmerge") then
+		if strFind(command,"unmerge") then
 			local afk = IsTeamAfk(GetTeamID(playerID))
-			spEcho("team is afk: " .. tostring(afk))
-			if controlledPlayers[playerID] and not afk then
+			DebugEcho("team is afk: " .. tostring(afk))
+			if not afk and #spGetPlayerList(playerID) > 1 then
 				UnmergePlayer(playerID)
 				return
 			else
-				spEcho("[Commshare] " .. playerID .. "(" .. name .. ") isn't on a squad!")
+				DebugEcho("[Commshare] " .. playerID .. "(" .. name .. ") is afk/not in a squad!")
 				return
 			end
 		end
 
 		if type(targetID) ~= "number" then
 			return
 		end
-
+		DebugEcho("[Commshare] Command: " .. tostring(command) .. " from " .. playerID)
 		-- Do commands --
 		if strFind(command, "invite") then
 			SendInvite(playerID, targetID)
 			if invites[playerID] and invites[playerID][targetID] and invites[targetID] and invites[targetID][playerID] then
 				AcceptInvite(playerID,targetID)
 			end
+		elseif command:find("playerchanged") then -- hack in remerging. this is sent
+			DebugEcho("[Commshare] Playerchanged: " .. targetID)
+			local name, active, spectator, teamID = spGetPlayerInfo(targetID)
+			local commshareID = spGetPlayerRulesParam(targetID, "commshare_team_id")
+			DebugEcho("playerstates: " .. tostring(playerstates[targetID] == nil) .. "\nSpectator: " .. tostring(spectator))
+			if playerstates[targetID] == nil and not spectator then -- this player has commshared or changed state.
+				DebugEcho("[Commshare] generated playerstate table.")
+				playerstates[targetID] = {active = active, spectator = spectator, teamid = teamID}
+			elseif not spectator then
+				DebugEcho("Commshare: PlayerChange: " .. name .."(ID: " .. targetID ..")\nActive: " .. tostring(playerstates[playerID].active) .. "->" .. tostring(active) .. "\nSpectator: " .. tostring(playerstates[playerID].spectator) .. "->" .. tostring(spectator) .."\nMergeID: " .. tostring(commshareID))
+				if active ~= playerstates[targetID].active and active and commshareID then -- this player has reconnected.
+					MergePlayer(targetID,commshareID)
+					DebugEcho("Commshare: Remerged " .. name)
+				end
+				playerstates[targetID].active = active
+				playerstates[targetID].spectator = spectator
+				playerstates[targetID].teamid = teamID
+			elseif spectator and playerstates[targetID] then -- this player resigned
+				DebugEcho("Commshare: Disposing of " .. name)
+				deadplayers.count = deadplayers.count + 1
+				deadplayers.players[targetID] = true
```

Comment:
see the linked suggestions above
```suggestion
				deadplayers[deadplayersCount] = targetID
				deadplayersCount = deadplayersCount + 1
```